src/instrumenter/instrument.ts

```diff
@@ -978,15 +1017,31 @@ function insertVarsStruct(
 }
 
 function getCheckStateInvsFuncs(contract: ContractDefinition): FunctionDefinition {
-    return single(contract.vFunctions.filter((fn) => fn.name === CHECK_STATE_INVS_FUN));
+    return single(
+        contract.vFunctions.filter(
+            (fn) =>
+                fn.name === CHECK_STATE_INVS_FUN || fn.name.slice(0, -2) === CHECK_STATE_INVS_FUN
+        )
+    );
 }
 
 function isPublic(fn: FunctionDefinition): boolean {
     return [FunctionVisibility.Default, FunctionVisibility.Public].includes(fn.visibility);
 }
 
 function getInternalCheckInvsFun(contract: ContractDefinition): string {
-    return `__scribble_${contract.name}_check_state_invariants_internal`;
+    const allNames = getAllNames(contract);
+    const funcName = `__scribble_${contract.name}_check_state_invariants_internal`;
+    if (!allNames.has(funcName)) {
+        return funcName;
+    }
+
+    let idx = 0;
+    while (allNames.has(`${funcName}_${idx}`)) {
+        idx++;
+    }
```

Comment:
I see that we have several occurances of the same logic within different functions. While I do understand the reasoning and approach, I also believe that we can dedicated utility function to reduce code duplication. WDYT?